### ISSUE TITLE
Look at replacing StringIO usage with simple Reader implementation

### ISSUE THREAD

**Issue by robertbartel** [NOAA-OWP/DMOD#609]
See [here](https://github.com/NOAA-OWP/DMOD/pull/566#discussion_r1592813643) for original suggestion.

In short, consider moving to a custom implementation of the _Reader_ protocol (or what [Pandas would refer to](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html) as a "file-like object") in places where _StringIO_ is used.

**Comment by robertbartel** [comment]
I am at least a bit curious the degree to which this is preferable to just using _StringIO_.  Wouldn't mind a bit of discussion and references here if possible before we proceed with anything.

**Comment by christophertubbs** [comment]
This is one of those things that makes my head tilt like a dog. Why not use a `BytesIO` when it's a `bytes` object?

**Comment by robertbartel** [comment]
> This is one of those things that makes my head tilt like a dog. Why not use a BytesIO when it's a bytes object?

Probably only because I wasn't thinking too hard about it, and the Pandas documentation for [`read_csv`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html) specifically cites _StringIO_ as an example of a file-like.  So it put a valid solution in my head, and I just went with it.

I probably should change that in the #566 review (and then update the issue title and description here; I think the general question probably still applies).

**Comment by aaraney** [comment]
Yeah, I concur, @robertbartel in regard to switching to `BytesIO`. I must have had a one tracked mind when reviewing! `BytesIO` didn't come to mind.
